### PR TITLE
test(ios-agent): close the three coverage gaps earlier reviews wrote down

### PR DESCRIPTION
**Pull request by jo-duchan**
## Summary

Three coverage gaps that earlier reviews found, recorded in `.work/`, and deferred. Each had a
mutation that survived the whole suite before this.

- **`O_NOFOLLOW`** in `readIfProviderWrote`. Deleting it left all 510 tests green, and it is what stops
  a symlink at the fallback path handing the class a file the provider never wrote.
- **The `file.mode & 0o022` half** of the same guard — the term that refuses a root-owned state file
  anyone can rewrite.
- **The descriptor-scan bound** in the injected library, which was the open question #763 dropped
  without deciding either way.

**The review overturned this PR's central claim.** It said both guard halves were only observable as
root. That is true of the file-mode half and false of `O_NOFOLLOW`: the guard stats the *link's*
directory, so putting the link somewhere only its owner can write makes the ownership half never run,
and refusing to follow becomes the only thing that decides — with a target the suite can own. The
comment asserting otherwise was silently assuming the link sat in a world-writable directory. A
non-root case covers it now and turns red on any machine when the flag goes.

The root-only case stays, because it covers the other side, and the two of them run under `sudo` on
the ubuntu runner. `vitest -t` exits 0 when its filter matches nothing and when every match skips, so
that step asserts two tests actually passed rather than trusting the exit code.

`tf_fd_scan_bound` also turned out to truncate: `rlim_t` is 64-bit, a finite limit above `INT_MAX`
passes the caller's `!= RLIM_INFINITY` test, and `(int)` of it is negative — so the scan walked
nothing while reporting that it had not trimmed. It decides in 64 bits now. The first attempt at that
clamp trimmed before asking whether it had trimmed, which answers "no" every time; the table caught it.

26 mutations, all killed.

## Checklist

- [x] Tests written and passing
- [x] No `any`
- [x] Interface changes land in `agent-core` first — n/a
- [x] No sensitive info (tokens, paths, credentials)

## Related `.work/` docs

- `.work/2026-09-05-netfilter-state-guard-coverage-todo.md` — the two guard gaps, now closed, with the
  correction to what "root only" meant.
- `.work/2026-09-07-nethook-pure-logic-tests-plan.md` — its open question 4, answered.
- `.work/reviews/test__close-recorded-coverage-gaps.md` — one channel, five findings, none deferred.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved state-file validation by rejecting symlinked files and unsafe root-owned, world-writable files.
  * Added bounded descriptor scanning with reliable fallback behavior when system limits are unavailable or invalid.
  * Updated the shipped network-hook package checksums.

* **Tests**
  * Expanded coverage for state-file safety and descriptor-scan boundary conditions, including root-only scenarios.

### CODE REVIEW

File: scripts/__tests__/nethookDecisions.test.mjs [hunk 1]
```diff
@@ -137,6 +137,34 @@ describe('which process gets hooked', () => {
...
+    [1, 256, 256, 0, 'a limit below the cap is used as it stands'],
+    [1, 8192, 8192, 0, 'exactly the cap is not a cap'],
+    [1, 8193, 8192, 1, 'one over is'],
+    [1, 1048576, 8192, 1, 'and so is a limit of the size that made this necessary'],
+    [0, 999999, 1024, 0, 'an unreadable or infinite limit falls back to 1024, not to the huge value'],
+    // **Above `INT_MAX`, which is where the cast used to turn the bound negative.** A negative bound
+    // makes the caller's `for (int fd = 0; fd < max; …)` walk nothing while `capped` says it did not
+    // trim — no connection cut and no line saying why. The table stopped one row short of it.
+    [1, 2147483648, 8192, 1, 'a limit above INT_MAX still clamps, and still says it clamped'],
+    [1, 9223372036854775806, 8192, 1, 'and so does one just under RLIM_INFINITY'],
```
[the remark this answers is not in the document]
Comment (jo-duchan) [reply scripts/__tests__/nethookDecisions.test.mjs]: Taken, and the rounding is worse than one place: `String(9223372036854775806)` renders `9223372036854776000`, which is not merely imprecise but **above** `RLIM_INFINITY` rather than one below it. So the row labelled "just under RLIM_INFINITY" was feeding a value the caller's `!= RLIM_INFINITY` test rejects, while claiming to cover the largest one it lets through — the reachable boundary stayed untested.

It is a string now. Worth saying plainly that this buys no killing power: both the reachable boundary and the unreachable value clamp to 8192 with `capped=1`, so the assertion passed either way and no mutation distinguishes them. What it buys is the row testing what its name says, which is the entire reason that row exists — a table of boundaries whose labels do not match its inputs is the shape this repository keeps finding on the other side of a green run.
